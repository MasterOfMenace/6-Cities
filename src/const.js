export const OfferRenderType = {
  MAIN: `main`,
  NEIGHBORHOOD: `neighborhood`
};

export const MonthNames = [
  `January`,
  `February`,
  `Mart`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`
];
