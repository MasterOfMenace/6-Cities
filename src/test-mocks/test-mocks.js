export const cities = [
  {
    name: `Amsterdam`,
    location: [52.38333, 4.9],
    zoom: 12
  },
  {
    name: `Paris`,
    location: [48.85657, 2.35186],
    zoom: 12
  },
  {
    name: `Cologne`,
    location: [50.93685, 6.9625],
    zoom: 12
  },
  {
    name: `Brussels`,
    location: [50.85300, 4.43792],
    zoom: 12
  },
  {
    name: `Hamburg`,
    location: [53.53539, 9.99371],
    zoom: 12
  },
  {
    name: `Dusseldorf`,
    location: [51.22977, 6.79138],
    zoom: 12
  },
];

export const offers = [
  {
    id: 1,
    city: {
      name: cities[0].name
    },
    title: `Offer1`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 100,
    rating: 4,
    type: `Private room`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 2,
    city: {
      name: cities[1].name
    },
    title: `Offer2`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 200,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 3,
    city: {
      name: cities[2].name
    },
    title: `Offer3`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 300,
    rating: 4,
    type: `Hostel`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 4,
    city: {
      name: cities[3].name
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 5,
    city: {
      name: cities[4].name
    },
    title: `Offer1`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 100,
    rating: 4,
    type: `Private room`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 6,
    city: {
      name: cities[5].name
    },
    title: `Offer2`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 200,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 7,
    city: {
      name: cities[1].name
    },
    title: `Offer3`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 300,
    rating: 4,
    type: `Hostel`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
  {
    id: 8,
    city: {
      name: cities[2].name
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  },
];

export const reviews = [
  {
    author: {
      avatar: `author-avatar-1`,
      id: 1,
      isPro: false,
      name: `author-1`
    },
    id: 1,
    rating: 3,
    text: `Review text 1`,
    date: `date`
  },
  {
    author: {
      avatar: `author-avatar-2`,
      id: 2,
      isPro: false,
      name: `author-2`
    },
    id: 2,
    rating: 5,
    text: `Review text 2`,
    date: `date`
  }
];

export const neighbors = [
  {
    id: 4,
    city: {
      name: `City`,
      location: [52.38333, 4.9]
    },
    title: `Offer4`,
    isFavorite: false,
    isPremium: true,
    previewImage: `img/apartment-01.jpg`,
    images: [`img/apartment-01.jpg`, `img/apartment-01.jpg`],
    description: `Description`,
    goods: [`good1`, `good2`],
    price: 400,
    rating: 4,
    type: `Apartment`,
    maxAdults: 2,
    bedrooms: 1,
    location: [52.369553943508, 4.85309666406198],
    host: {
      avatarUrl: `avatar`,
      id: 1,
      isPro: true,
      name: `host-name`
    }
  }
];

export const userInfo = {
  avatarUrl: `/img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  name: `Oliver.conner`
};
