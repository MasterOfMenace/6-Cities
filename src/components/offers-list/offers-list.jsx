import React from 'react';
import PropTypes from 'prop-types';
import {OfferRenderType} from '../../const.js';
import OfferCard from '../offer-card/offer-card.jsx';

class OffersList extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      offer: null
    };
  }

  render() {
    const {offers, titleClickHandler, type} = this.props;

    const className = type === OfferRenderType.MAIN ? `cities__places-list places__list tabs__content` : `near-places__list places__list`;

    return (
      <div className={className}>
        {offers.map((offer, index) => (
          <OfferCard
            key={index}
            offer={offer}
            onMouseOver={() => {
              this.setState({
                offer: offer.id
              });
            }}
            onMouseLeave={() => {
              this.setState({
                offer: null
              });
            }}
            titleClickHandler={titleClickHandler}
            type={type}/>
        ))}
      </div>
    );
  }
}

OffersList.propTypes = {
  type: PropTypes.oneOf([OfferRenderType.MAIN, OfferRenderType.NEIGHBORHOOD]).isRequired,
  offers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    location: PropTypes.arrayOf(PropTypes.number).isRequired,
    reviews: PropTypes.arrayOf(PropTypes.shape({
      author: PropTypes.string.isRequired,
      avatar: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
      time: PropTypes.string.isRequired
    })).isRequired
  })).isRequired,
  titleClickHandler: PropTypes.func.isRequired
};

export default OffersList;
