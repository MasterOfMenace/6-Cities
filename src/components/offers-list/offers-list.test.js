import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import {offers as mockOffers} from '../../test-mocks/test-mocks.js';
import {OffersList} from './offers-list';
import {OfferRenderType} from '../../const.js';
import history from '../../history.js';

it(`Правильное отображение компонента OffersList с OfferRenderType.MAIN`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <OffersList
            type={OfferRenderType.MAIN}
            offers={mockOffers}
            onMouseOver={()=>{}}
            onMouseLeave={()=>{}}
            titleClickHandler={()=>{}}
            onFavoriteButtonClick={()=>{}}
            isAuth={true}/>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Правильное отображение компонента OffersList с OfferRenderType.NEIGHBORHOOD`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <OffersList
            type={OfferRenderType.NEIGHBORHOOD}
            offers={mockOffers}
            onMouseOver={()=>{}}
            onMouseLeave={()=>{}}
            titleClickHandler={()=>{}}
            onFavoriteButtonClick={()=>{}}
            isAuth={true}/>
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
