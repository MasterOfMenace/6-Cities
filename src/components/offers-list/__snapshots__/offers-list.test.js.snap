// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Правильное отображение компонента OffersList с OfferRenderType.MAIN 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer2
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            300
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer3
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Hostel
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            400
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer4
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;

exports[`Правильное отображение компонента OffersList с OfferRenderType.NEIGHBORHOOD 1`] = `
<div
  className="near-places__list places__list"
>
  <article
    className="near-places__card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer2
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            300
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer3
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Hostel
      </p>
    </div>
  </article>
  <article
    className="near-places__card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="near-places__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            400
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button   button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
          onClick={[Function]}
        >
          Offer4
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
